backend/src/main/scala/cromwell/backend/standard/callcaching/StandardCacheHitCopyingActor.scala

```diff
@@ -129,28 +128,36 @@ abstract class StandardCacheHitCopyingActor(val standardParams: StandardCacheHit
 
   when(Idle) {
     case Event(CopyOutputsCommand(simpletons, jobDetritus, returnCode), None) =>
-      val sourceCallRootPath = lookupSourceCallRootPath(jobDetritus)
-
-      val processed = for {
-        (callOutputs, simpletonCopyPairs) <- processSimpletons(simpletons, sourceCallRootPath)
-        (destinationDetritus, detritusCopyPairs) <- processDetritus(jobDetritus)
-      } yield (callOutputs, destinationDetritus, simpletonCopyPairs ++ detritusCopyPairs)
-
-      processed match {
-        case Success((callOutputs, destinationDetritus, detritusAndOutputsIoCommands)) =>
-          duplicate(ioCommandsToCopyPairs(detritusAndOutputsIoCommands)) match {
-            case Some(Success(_)) => succeedAndStop(returnCode, callOutputs, destinationDetritus)
-            case Some(Failure(failure)) => failAndStop(failure)
-            case None =>
-
-              if (detritusAndOutputsIoCommands.nonEmpty) {
-                detritusAndOutputsIoCommands foreach { sendIoCommand(_) }
-                
-                // Add potential additional commands to the list
-                val allCommands = List(detritusAndOutputsIoCommands) ++ additionalIoCommands(simpletons, callOutputs, jobDetritus, destinationDetritus)
-                
-                goto(WaitingForIoResponses) using Option(StandardCacheHitCopyingActorData(allCommands, callOutputs, destinationDetritus, returnCode))
-              } else succeedAndStop(returnCode, callOutputs, destinationDetritus)
+
+      // Try to make a Path of the callRootPath from the detritus
+      lookupSourceCallRootPath(jobDetritus) match {
+        case Success(sourceCallRootPath) =>
+          
+          // process simpletons and detritus to get updated paths and corresponding IoCommands
+          val processed = for {
+            (destinationCallOutputs, simpletonIoCommands) <- processSimpletons(simpletons, sourceCallRootPath)
+            (destinationDetritus, detritusIoCommands) <- processDetritus(jobDetritus)
+          } yield (destinationCallOutputs, destinationDetritus, simpletonIoCommands ++ detritusIoCommands)
+
+          processed match {
+            case Success((destinationCallOutputs, destinationDetritus, detritusAndOutputsIoCommands)) =>
+              duplicate(ioCommandsToCopyPairs(detritusAndOutputsIoCommands)) match {
+                  // Use the duplicate override if exists
+                case Some(Success(_)) => succeedAndStop(returnCode, destinationCallOutputs, destinationDetritus)
+                case Some(Failure(failure)) => failAndStop(failure)
+                  // Otherwise send the first round of IoCommands (file outputs and detritus) if any
+                case None if detritusAndOutputsIoCommands.nonEmpty =>
+                    detritusAndOutputsIoCommands foreach { sendIoCommand(_) }
```

Comment:
if `sendIoCommand` was refactored to have two parameter lists I think this could then `foreach sendIoCommand`